src/idna.rs

```diff
@@ -31,15 +27,11 @@ impl Idna {
     ///
     /// ```
     /// use ada_url::Idna;
-    /// assert_eq!(Idna::ascii("meßagefactory.ca"), "xn--meagefactory-m9a.ca");
+    /// assert_eq!(Idna::ascii("meßagefactory.ca").as_ref(), "xn--meagefactory-m9a.ca");
     /// ```
     #[must_use]
-    pub fn ascii(input: &str) -> &str {
-        unsafe {
-            let out = ffi::ada_idna_to_ascii(input.as_ptr().cast(), input.len());
-            let slice = core::slice::from_raw_parts(out.data.cast(), out.length);
-            core::str::from_utf8_unchecked(slice)
-        }
+    pub fn ascii(input: &str) -> ffi::ada_owned_string {
```

Comment:
ada_owned_string is a private API. if the C API changes, we need to also change our implementation, so I'm against exposing this